### PR TITLE
Removed 'az subscription set'  and bumped the 4.7 version to 4.7.24

### PR DESCRIPTION
**Pull request by mwoodson**
### Which issue this PR addresses:

- Fixes deployment of gateway VMSS, which is getting hung up on setting subscription in canary regions.
- bumped the version to 4.7.

### What this PR does / why we need it:

Remove unnecessary set of subscription.  We deploy to a resource group, and never reuse names for RG's for RP / Gateway.  It's going to be in the right subscription.

Bumping the version to 4.7.24 is needed to resolve bug https://bugzilla.redhat.com/show_bug.cgi?id=1916501.  4.6 Z is needed as well, but not in stable yet.

### Test plan for issue:

Deploy to INT and Canary

### Is there any documentation that needs to be updated for this PR?

Comment where removed indicating why.

### CODE REVIEW

File: pkg/deploy/generator/resources_gateway.go [hunk 1]
```diff
@@ -318,7 +318,11 @@ EOF
 
 export AZURE_CLOUD_NAME=$AZURECLOUDNAME
 az login -i --allow-no-subscriptions
-az account set -s "$SUBSCRIPTIONID"
+
+# Running "az account set" is not working in the pipeline.
+# we are running VM ScaleSet in a named ResourceGroup, it should
+# be OK to to remove this set subscription
+# az account set -s "$SUBSCRIPTIONID"
```
Comment (m1kola) [inline pkg/deploy/generator/resources_gateway.go:325]: Why is it failing?

> The subscription of 'blah' doesn't exist in cloud 'AzureCloud'.

I don't get it: `pkg/deploy/generator/resources_rp.go` has exactly the same piece. Why we did not have an issue there?

**Review by jewzaam — APPROVED** [review]

File: pkg/deploy/generator/resources_gateway.go (hunk 1 above)
Comment (jewzaam) [reply pkg/deploy/generator/resources_gateway.go:325]: We don't know.  This was introduced for the RP earlier this year.  Gateway deployment to eastus2euap was fine without this line and now it is also failing with this as well.
```
++ export AZURE_CLOUD_NAME=AzureCloud
++ AZURE_CLOUD_NAME=AzureCloud
++ az login -i --allow-no-subscriptions
++ az account set -s my-awesome-subscription-id
ERROR: The subscription of 'my-awesome-subscription-id' doesn't exist in cloud 'AzureCloud'.
``` 

Where the subscription ID is the canary sub id.  Perhaps there's something in the RG setup that is missed.  Regardless, this isn't a necessary step as this code is running on the VM's in a VMSS deployed to a RG which is in the subscription.

**Review by jewzaam — APPROVED** [review]

**Pull request merged**